src/routes/members/gigs/[gig_id([0-9]+)]/edit.svelte

```diff
@@ -0,0 +1,1133 @@
+    advertise,
+    admins_only,
+    allow_signups: typeCode !== "calendar" && allow_signups,
+    food_provided: typeCode !== "calendar" && food_provided,
+    notes_admin: notes_admin && notes_admin.trim(),
+    notes_band: notes_band && notes_band.trim(),
+    summary: summary && summary.trim(),
+    arrive_time: typeCode === "calendar" ? DateTime.fromISO(arrive_time_date) : arrive_time,
+    finish_time: typeCode === "calendar" ? DateTime.fromISO(finish_time_date) : finish_time,
+    time: time || null,
+    contacts,
+    lineup: [],
+    venue,
+    finance: finance && finance.trim(),
+    finance_deposit_received,
+    finance_payment_received,
+    finance_caller_paid,
+    quote_date,
+  };
+
+  let validityFields = {};
+
+  const checkValid = (node, options) => {
+    if (options.bothPresent) {
+      if (!validityFields[options.bothPresent.id]) {
+        validityFields[options.bothPresent.id] = [];
+      }
+      validityFields[options.bothPresent.id].push(node);
+    }
+    const changeHandler = () => {
+      for (let key of Object.keys(options.validityErrors)) {
+        if (node.validity[key]) {
+          node.setCustomValidity(options.validityErrors[key]);
+          return;
+        }
+      }
+
+      node.setCustomValidity("");
+
+      if (options.bothPresent) {
+        let presence = validityFields[options.bothPresent.id].map((field) => field.value.length > 0);
+        if (!presence.every((x) => x) && !presence.every((x) => !x)) {
+          for (let field of validityFields[options.bothPresent.id]) {
+            if (!field.value.length) {
+              field.setCustomValidity(options.bothPresent.error);
+            }
+          }
+          return;
+        }
+        for (let field of validityFields[options.bothPresent.id]) {
+          if (field.validationMessage === options.bothPresent.error) {
+            field.setCustomValidity("");
+            field.dispatchEvent(new Event("change"));
+          }
+        }
+      }
+    };
+
+    node.addEventListener("change", changeHandler);
+
+    return {
+      destroy() {
+        node.removeEventListener("change", changeHandler);
+      },
+    };
+  };
+
+  function unloadIfSaved(e) {
+    if (saved || window.Cypress) {
+      delete e["returnValue"];
+    } else {
+      e.preventDefault();
+      e.returnValue = "";
+    }
+  }
+
+  function selectVenueSearch(e) {
+    venue_id = e.detail.id;
+    venueListElement.focus();
+  }
+
+  function selectClientSearch(e) {
+    selectedClient = e.detail.id;
+    clientListElement.focus();
+  }
+
+  function selectCallerSearch(e) {
+    selectedCaller = e.detail.id;
+    callerListElement.focus();
+  }
+
+  function newVenue() {
+    venue = {};
+    editVenue();
+  }
+
+  function newSubvenue() {
+    venue = { name: venues.find((venue) => venue.id === venue_id).name };
+    editVenue();
+  }
+
+  function editVenue() {
+    editingSubvenue = venue && Object.keys(venue).length === 1; // Only name set for this
```

Comment:
what is this?

Reply:
(comment is useful but i dont know what this refers to) "this is a venue with only a name"?